app/routers/systems.py

```diff
@@ -1,45 +1,62 @@
-from uuid import UUID
+from typing import Annotated
 
-from fastapi import APIRouter, status
+from fastapi import APIRouter, Depends, HTTPException, status
 from fastapi_pagination.limit_offset import LimitOffsetPage
 
-from app.schemas import SystemCreate, SystemRead, SystemUpdate
+from app.auth.auth import UNAUTHORIZED_EXCEPTION
+from app.db.handlers import NotFoundError
+from app.db.models import System
+from app.dependencies import CurrentAuthContext, SystemId, SystemRepository
+from app.schemas import SystemCreate, SystemRead, SystemUpdate, from_orm
 
 router = APIRouter()
 
 
+async def fetch_system_or_404(id: SystemId, system_repo: SystemRepository) -> System:
+    try:
+        return await system_repo.get(id=id)
```

Comment:
We should put the following logic here:
```suggestion
        extra_conditions = []
        if auth_context.account.type == AccountType.AFFILIATE:
            extra_conditions.append(System.owner == auth_context.account)
        return await system_repo.get(id=id, *extra_conditions)
```

Reply:
Added, thanks